internal/home/service.go

```diff
@@ -681,3 +687,55 @@ rc_bg=YES
 
 rc_cmd $1
 `
+
+// secureBinary is used before service.Install(). This function protects AdGuardHome from
+// privilege escalation vulnerabilities caused by writable files
+func secureBinary() error {
+	switch runtime.GOOS {
+	case "windows":
+		// TODO: support windows service support securely
+		// Set file owner to admin/system and public permissions read-only
+		return errors.Error("you currently cannot install adguardhome as a service on window")
+	default:
+		return secureBinaryUnix()
+	}
+}
+
+func secureBinaryUnix() error {
+	// Installalation can only be completed with root privileges, so check and handle if not
+	if os.Getuid() != 0 {
+		return errors.Error("permission denied. Root privileges required")
+	}
+
+	// Get current file path
+	binary := os.Args[0]
+
+	// Change owner to root:root
+	err := os.Chown(binary, 0, 0)
+	if err != nil {
+		return err
+	}
+
+	// Set permissions to root(read,write,exec), group(read,exec), public(read)
+	// This combined with changing the owner make the file undeletable without root privlages
+	// UNLESS THE PARENT FOLDER IS WRITABLE!
+	if err := os.Chmod(binary, 0755); err != nil {
+		return err
+	}
+
+	// Move binary to the PATH in a folder which is read-only to non root users
+	if err := os.Rename(binary, "/usr/bin/AdGuardHome"); err != nil {
```

Comment:
Hard-coding the path here is problematic

Reply:
Would you prefer it be a CLI option called "install-dir" defined in `internal\home\options.go`  `var cmdLineOpts = []cmdLineOpt{{` with a default value of "/usr/bin".  Then the dir variable will have the executable name appended set as  "AdGuardHome" (which can be defined as a constant).

 Alternatively I can make it a constant defined at the top of the file.